scripts/build-and-deploy-private-ip.sh

```diff
@@ -0,0 +1,135 @@
+#! /bin/bash
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# variables required for below commands to properly build and deploy GroupSync
+######################## DEPLOYMENT variables ########################
+export PROJECT_ID="" # project ID of project in which you want to deploy the service within
+export REGION="" # Google Cloud region to deploy GroupSync in
+
+######################## Service Account variables ########################
+export SERVICE_ACCOUNT_NAME="" # name of service account to create and use with GroupSync
+export SERVICE_ACCOUNT_EMAIL="$SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com" # email of service account to deploy Cloud Run with
+
+######################## PRIVATE IP / SHARED VPC variables ########################
+export HOST_PROJECT_ID="" # project ID of Shared VPC host project (optional)
+export CONNECTOR_NAME="" # name to be given to Serverless VPC Access Connector
+export SUBNET="" # the name of an unused /28 subnet for Serverless VPC Access Connector
+
+######################## Cloud Scheduler variables ########################
+export PATH_TO_JSON="" # file path to JSON payload containing instance-to-group mappings for Cloud Scheduler
+export SCHEDULE="*/10 * * * *" # schedule how often GroupSync Cloud Scheduler is called (defaults to 10 mins)
+
+# load IAM Groups and Cloud SQL Instance names from JSON payload
+IAM_GROUPS=`cat $PATH_TO_JSON | jq '.iam_groups' | tr -d '[],"'`
+SQL_INSTANCES=`cat $PATH_TO_JSON | jq '.sql_instances' | tr -d '[],"'`
+
+# check if required variables are set, otherwise give error and exit
+declare -a vars=(PROJECT_ID REGION PATH_TO_JSON IAM_GROUPS SQL_INSTANCES CONNECTOR_NAME HOST_PROJECT_ID SUBNET)
```

Comment:
Would this be simpler as the following?

``` sh
for var_name in A B C D
do
  if [[ -z "${var_name}" ]]; then
    # not defined
  fi
done
```

https://stackoverflow.com/a/39296583/927514

Reply:
This didn't seem to work for me, will leave it as is for now.